rewrite-graphql/src/main/java/org/openrewrite/graphql/internal/GraphQlPrinter.java

```diff
@@ -0,0 +1,1083 @@
+            visitValue(inputValueDef.getDefaultValue(), p);
+        }
+        visitList(inputValueDef.getDirectives(), p);
+        afterSyntax(inputValueDef, p);
+        return inputValueDef;
+    }
+
+    @Override
+    public GraphQl visitInterfaceTypeDefinition(GraphQl.InterfaceTypeDefinition interfaceType, PrintOutputCapture<P> p) {
+        beforeSyntax(interfaceType, p);
+        if (interfaceType.getDescription() != null) {
+            visitStringValue(interfaceType.getDescription(), p);
+            p.append(" ");
+        }
+        p.append("interface");
+        visitName(interfaceType.getName(), p);
+        visitImplementsInterfaces(interfaceType.getImplementsInterfaces(), p);
+        visitList(interfaceType.getDirectives(), p);
+        if (interfaceType.getFieldsPrefix() != null) {
+            printSpace(interfaceType.getFieldsPrefix(), p);
+            if (interfaceType.getFields() != null && !interfaceType.getFields().isEmpty()) {
+                visitContainer("{", interfaceType.getFields(), "}", null, p);
+            } else {
+                p.append("{}");
+            }
+        }
+        afterSyntax(interfaceType, p);
+        return interfaceType;
+    }
+
+    @Override
+    public GraphQl visitUnionTypeDefinition(GraphQl.UnionTypeDefinition unionType, PrintOutputCapture<P> p) {
+        beforeSyntax(unionType, p);
+        if (unionType.getDescription() != null) {
+            visitStringValue(unionType.getDescription(), p);
+            p.append(" ");
+        }
+        p.append("union");
+        visitName(unionType.getName(), p);
+        visitList(unionType.getDirectives(), p);
+        if (unionType.getMemberTypes() != null && !unionType.getMemberTypes().isEmpty()) {
+            p.append(" =");
+            for (int i = 0; i < unionType.getMemberTypes().size(); i++) {
+                if (i > 0) {
+                    p.append(" |");
+                }
+                visitNamedType(unionType.getMemberTypes().get(i), p);
+            }
+        }
+        afterSyntax(unionType, p);
+        return unionType;
+    }
+
+    @Override
+    public GraphQl visitEnumTypeDefinition(GraphQl.EnumTypeDefinition enumType, PrintOutputCapture<P> p) {
+        beforeSyntax(enumType, p);
+        if (enumType.getDescription() != null) {
+            visitStringValue(enumType.getDescription(), p);
+            p.append(" ");
+        }
+        p.append("enum");
+        visitName(enumType.getName(), p);
+        visitList(enumType.getDirectives(), p);
+        visitContainer(" {", enumType.getValues(), "}", null, p);
+        afterSyntax(enumType, p);
+        return enumType;
+    }
+
+    @Override
+    public GraphQl visitEnumValueDefinition(GraphQl.EnumValueDefinition enumValueDef, PrintOutputCapture<P> p) {
+        beforeSyntax(enumValueDef, p);
+        if (enumValueDef.getDescription() != null) {
+            visitStringValue(enumValueDef.getDescription(), p);
+            p.append(" ");
```

Comment:
Printing single space literals like this is always fishy. Probably means what we should look into making something a `LeftPadded` or `RightPadded` somewhere.